src/hyperion/parameters/gridscan.py

```diff
@@ -0,0 +1,263 @@
+from __future__ import annotations
+
+from typing import Any
+
+import numpy as np
+from dodal.devices.detector import DetectorParams
+from dodal.devices.fast_grid_scan import GridScanParams
+from dodal.devices.zebra import (
+    RotationDirection,
+)
+from pydantic import validator
+from scanspec.core import Path as ScanPath
+from scanspec.specs import Line
+
+from hyperion.external_interaction.ispyb.ispyb_dataclass import (
+    GridscanIspybParams,
+)
+from hyperion.parameters.components import (
+    DiffractionExperiment,
+    OptionalGonioAngleStarts,
+    RotationAxis,
+    TemporaryIspybExtras,
+    WithSample,
+    WithScan,
+    XyzAxis,
+    XyzStarts,
+)
+from hyperion.parameters.constants import CONST
+
+
+class GridCommon(DiffractionExperiment, OptionalGonioAngleStarts, WithSample):
+    grid_width_um = CONST.PARAM.GRIDSCAN.WIDTH_UM
+    exposure_time_s: float = CONST.PARAM.GRIDSCAN.EXPOSURE_TIME_S
+    use_roi_mode: bool = CONST.PARAM.GRIDSCAN.USE_ROI
+    transmission_frac: float = 1
+    # field rather than inherited to make it easier to track when it can be removed:
+    ispyb_extras: TemporaryIspybExtras = TemporaryIspybExtras()
+
+
+class GridScanWithEdgeDetect(GridCommon, WithSample): ...
+
+
+class PinTipCentreThenXrayCentre(GridCommon):
+    tip_offset_um: float = 0
+
+
+class RobotLoadThenCentre(GridCommon, WithSample): ...
+
+
+class SpecifiedGridScan(GridCommon, XyzStarts, WithScan, WithSample):
+    @property
+    def detector_params(self):
+        detector_params = {
+            "expected_energy_ev": self.demand_energy_ev,
+            "exposure_time": self.exposure_time_s,
+            "directory": str(
+                self.visit_directory / "xraycentring" / str(self.sample_id)
+            ),
+            "prefix": self.file_name,
+            "detector_distance": self.detector_distance_mm,
+            "omega_start": self.omega_start_deg,
+            "omega_increment": 0,
+            "num_images_per_trigger": 1,
+            "num_triggers": self.num_images,
+            "use_roi_mode": self.use_roi_mode,
+            "det_dist_to_beam_converter_path": self.det_dist_to_beam_converter_path,
+            "run_number": self.run_number,
+            "trigger_mode": self.trigger_mode,
+        }
+        return DetectorParams(**detector_params)
+
+    @property
+    def ispyb_params(
+        self,
```

Comment:
Nit: Is `black` only putting this on a new line because of the comma? I think remove the comma...